datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -142,21 +142,11 @@ impl PhysicalExpr for ScalarFunctionExpr {
     }
 
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
-        // evaluate the arguments, if there are no arguments we'll instead pass in a null array
-        // indicating the batch size (as a convention)
-        let inputs = match self.args.is_empty() {
-            // If the function supports zero argument, we pass in a null array indicating the batch size.
-            // This is for user-defined functions.
-            // MakeArray support zero argument but has the different behavior from the array with one null.
```

Comment:
The reason why we need to add special case for `make_array` is that we need to differentiate empty array `[]` and single null array `[null]`, previously we avoid additional null array by filter out `make_array`. Now, I think it is even not a good idea to add specializaton to effect every functions, so remove it at all.

Reply:
So is the thinking `support_zero_argument` is no longer needed? Should that field be removed entirely now?

If all the existing tests pass this certainly seems like a reasonable special case to remove (maybe we could update the documentation on `ScalarUDFImpl` to explain how to handle funcitons that take zero arguments  -- namely to return a single row array (or ColumnarValue)

Reply:
I think we don't need it now, I don't have a strong opinion on whether to keep it or not.

> namely to return a single row array (or ColumnarValue)

functions that take 0 args are not restricted to return a single row array.  I think they can have a function that returns anything as long as is `Result<ColumnarValue>`

Reply:
Maybe we can remove `support_zero_argument `, add it back if needed.

Reply:
I think `support_zero_argument` can be handled in `signature check`, and actually I can't find a test that failed on `Expect {} function to take no parameters` easily. I think we have a good reason to remove `support_zero_argument` now.